src/MobileRedirection.php

```diff
@@ -78,6 +78,14 @@ public function register() {
 				add_filter( 'get_comments_link', [ $this, 'add_noamp_mobile_query_var' ] ); // For get_comments_link().
 				add_filter( 'respond_link', [ $this, 'add_noamp_mobile_query_var' ] ); // For comments_popup_link().
 			}
+
+			return;
+		}
+
+		// Show `Exit mobile version` link even mobile redirection is disabled and in Reader mode.
+		if ( AMP_Theme_Support::READER_MODE_SLUG === AMP_Options_Manager::get_option( Option::THEME_SUPPORT ) && $this->is_mobile_request() ) {
+			add_action( 'wp_head', [ $this, 'add_mobile_version_switcher_styles' ] );
+			add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
```

Comment:
@westonruter As you suggested in #5293, I have added `Exit mobile version` when the site is in reader mode and redirection is disabled. Can you review this specifically if this is the correct implementation?

Reply:
It looks like this is missing a couple things for legacy Reader mode:

https://github.com/ampproject/amp-wp/blob/40688e0592491bc85325deff48ee7236e753bdfe/src/MobileRedirection.php#L156
https://github.com/ampproject/amp-wp/blob/40688e0592491bc85325deff48ee7236e753bdfe/src/MobileRedirection.php#L187

Reply:
Also I guess:

https://github.com/ampproject/amp-wp/blob/40688e0592491bc85325deff48ee7236e753bdfe/src/MobileRedirection.php#L182-L183